packages/position/src/react/index.js

```diff
@@ -49,48 +49,52 @@ LeftOf.displayName = 'LeftOf'
 
 const Position = React.forwardRef((props, forwardedRef) => {
   const { target, position: positionFn, when } = props
+  const [shownOnce, setShownOnce] = React.useState(false)
   const [style, setStyle] = React.useState({ position: 'absolute' })
 
   const ref = React.useRef()
   React.useImperativeHandle(forwardedRef, () => ref.current)
 
   const child = React.Children.only(props.children)
-  const tetheredRef = React.useRef()
-  const tetheredEl = React.cloneElement(props.show, {
-    ref: tetheredRef,
+  const showRef = React.useRef()
+  const showEl = React.cloneElement(props.show, {
+    ref: showRef,
     style: { ...child.props.style, ...style }
   })
 
   const updateStyle = React.useCallback(() => {
-    if (!when) return
-
     const targetNode = target ? target.current : ref.current
-    if (!targetNode) return
+    if (!showRef.current || !targetNode) return
 
-    setTimeout(() => {
-      if (!tetheredRef.current) return
-      const nextStyle = positionFn(targetNode).styleFor(tetheredRef.current)
-      setStyle(nextStyle)
-    }, 1)
-  }, [positionFn, target, when])
+    const nextStyle = positionFn(targetNode).styleFor(showRef.current)
+    setStyle(nextStyle)
+    setShownOnce(true)
+  }, [positionFn, target])
 
-  React.useEffect(() => {
-    updateStyle()
-  }, [updateStyle])
+  useOnWindowResize(evt => updateStyle())
+  useOnWindowScroll(evt => {
+    const isInner = !showRef.current || showRef.current.contains(evt.target)
+    if (isInner) return
 
-  useOnWindowResize(updateStyle)
-  useOnWindowScroll(updateStyle)
+    updateStyle()
+  })
+  React.useEffect(() => {
+    if (!when) return
+    const timerId = delayUntilNextTick(() => updateStyle())
 
-  const inPortal = props.inNode
+    return () => clearTimeout(timerId)
+  }, [updateStyle, when])
 
   return (
     <React.Fragment>
       {target ? child : React.cloneElement(child, { ref })}
 
-      {when &&
-        (inPortal
-          ? ReactDOM.createPortal(tetheredEl, props.inNode)
-          : tetheredEl)}
+      {createUniversalPortal(
```

Comment:
Why always render in portal?

What tradeoffs does this represent?  For instance, is Position any less useful generally?

Reply:
using a portal helped simplify and fix some issues i was having. also i think a portal is the write call and we know that other's have been using it this way. i don't believe it makes it less useful but let me know if you have some scenarios i didn't think of